packages/frontend/src/components/HoursWithdrawal.vue

```diff
@@ -1,351 +1,528 @@
+                        {{ subItem.sum }}
+                      </div>
+                    </td>
+                    <td class="md-table-cell"></td>
+                  </tr>
+                </template>
+              </template>
+            </tbody>
+          </table>
+        </div>
+      </div>
+    </div>
+  </CenterColumnWrapper>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import { v4 as uuidv4 } from "uuid";
+import YellowButton from "./YellowButton.vue";
+import { isFloat } from "@/store/timeEntries";
+import Input from "./Input.vue";
+import moment, { Moment } from "moment";
+import { Store } from "vuex";
+import { State } from "../store/index";
+import CenterColumnWrapper from "@/components/CenterColumnWrapper.vue";
+import OvertimeVisualizer from "@/components/OvertimeVisualizer.vue";
+import { MappedOvertimeTransaction } from "../store/overtime";
+
+interface ValidationRule {
+  errorMessage: string;
+  validator: (hours: string, available: number) => boolean;
+}
+
+interface InternalTransaction {
+  id: number;
+  date: string;
+  hours: number;
+  type: string;
+  rate?: number;
+  sum?: number;
+  active?: boolean;
+  subItems?: InternalTransaction;
+}
+
+const rules: ValidationRule[] = [
+  {
+    errorMessage: "Skriv inn gyldig tall",
+    validator: (hours, _) => isFloat(hours as string),
+  },
+  {
+    errorMessage: "Antall timer må være større enn 0",
+    validator: (hours, _) => Number(hours) > 0,
+  },
+  {
+    errorMessage: "Kun utbetaling i halve timer",
+    validator: (hours, _) => Number(hours) % 0.5 === 0,
+  },
+  {
+    errorMessage: "Du kan ikke ta ut flere timer enn du har i banken",
+    validator: (hours, available) => Number(hours) <= available,
+  },
+];
+
+export default Vue.extend({
+  components: {
+    YellowButton,
+    Input,
+    CenterColumnWrapper,
+    OvertimeVisualizer,
+  },
+  data() {
+    return {
+      hours: "",
+      expandedTransaction: null,
+      transactions: [],
+      today: moment().format("YYYY-MM-DD"),
+      overtimeData: [],
+      holidayData: [],
+      holidaySubtractions: [],
+      unsubscribe: () => {},
+    };
+  },
+  computed: {
+    overtime(): number {
+      return this.$store.getters.getAvailableHours;
+    },
+    hoursText(): string {
+      return this.overtime === 1 ? "time" : "timer";
+    },
+    sortedTransactions(): InternalTransaction[] {
+      if (this.transactions) {
+        const toPayout = "Til utbetaling";
+        const transactions = (this.transactions as InternalTransaction[])
+          .sort((a, b) => {
+            let aDate = new Date(a.date);
+            let bDate = new Date(b.date);
+            if (a.active) return 0 - aDate.getTime();
+            return bDate.getTime() - aDate.getTime();
+          })
+          .map((transaction, index) => {
+            let newIdTrans = { ...transaction };
+            // Give new ids from index
+            newIdTrans.id = uuidv4();

```

Comment:
Av nysjerrighet. Hvorfor bruker du uuid her?

Reply:
Behov for key i listen. Får ikke id på annet enn "til utbetaling" fra backend. Må også generere egne id på aggregerte oppføringer. Anså uuid som greiest ruten til unike id, ettersom de uansett kun skal leve i kontekst av listevisningen.